src/inmanta/server/agentmanager.py

```diff
@@ -140,8 +140,8 @@ def get_depended_by(self) -> List[str]:
         assert isinstance(presession, SessionManager)
         presession.add_listener(self)
 
-    def new_session(self, session: protocol.Session) -> None:
-        self.add_background_task(self._register_session(session, datetime.now()))
+    async def new_session(self, session: protocol.Session) -> None:
+        await self._register_session(session, datetime.now())
```

Comment:
This seems risky: 
1. it can cause agent heartbeat to hang unexpectedly
2. it could cause the agents get_state call to be rejected ?